risc0/circuit/rv32im/src/lib.rs

```diff
@@ -39,7 +39,11 @@ pub use self::zirgen::CircuitImpl;
 
 pub const RV32IM_SEAL_VERSION: u32 = 1;
 
-pub const MAX_INSN_CYCLES: usize = 2000; // TODO(flaub): calculate actual value
+/// This number was picked by running `bigint2-analyze` on all the current bigint programs
+pub const MAX_INSN_CYCLES: usize = 25_000;
```

Comment:
from your worst-case estimator, i see the worst case as this:
```
risc0/bigint2/src/field/extfield_deg4_mul_256.blob
total = 19275
```
Where does the additional 5k cycles come from? Could this be `20_000` instead?

Reply:
I mainly wanted to leave a little room and not be so close to the max one right now. If we want to add some more bigint precompiles in the future they might go over this max if too low.